src/System Application/App/AI/src/Azure OpenAI/AOAIDeploymentsImpl.Codeunit.al

```diff
@@ -15,11 +16,15 @@ codeunit 7769 "AOAI Deployments Impl"
     InherentPermissions = X;
 
     var
+        Telemetry: Codeunit Telemetry;
         UnableToGetDeploymentNameErr: Label 'Unable to get deployment name, if this is a third party capability you must specify your own deployment name. You may need to contact your partner.';
         GPT4oLatestLbl: Label 'gpt-4o-latest', Locked = true;
         GPT4oPreviewLbl: Label 'gpt-4o-preview', Locked = true;
         GPT4oMiniLatestLbl: Label 'gpt-4o-mini-latest', Locked = true;
         GPT4oMiniPreviewLbl: Label 'gpt-4o-mini-preview', Locked = true;
+        DeprecatedDeployments: Dictionary of [Text, Date];
+        DeprecationDatesInitialized: Boolean;
+        DeprecationMessageLbl: Label 'Deployment %1 deprecated from %2. Check out codeunit 7768 AOAI Deployments', Comment = 'Telemetry message where %1 is the name of the deployment and %2 is the date of deprecation';
```

Comment:
Let's make the text a bit more dev friendly.
`We detected usage of the Azure OpenAI deployment "%1". This model is obsoleted starting %2 and the quality of your results might vary after that date. Check out codeunit 7768 AOAI Deployments to find the supported deployments.`